### ISSUE TITLE
Tables

### ISSUE THREAD

**Issue by lucyabb** [crocodic-studio/crudbooster#673]
I have 2 questions:

[1] Can I add a statement like "if status=0" to show only rows in a table where status is 0?  I want to do this to show a list of rows which are items that need approval.

[2] Is it possible to make some fields read only, so users can see but not edit them?

Thanks

**Comment by jdredondo** [comment]
1. Yes! You can use the method hook_query_index:
`/*
	    | ---------------------------------------------------------------------- 
	    | Hook for manipulate query of index result 
	    | ---------------------------------------------------------------------- 
	    | @query = current sql query 
	    |
	    */
	    public function hook_query_index(&$query) {
	        //Your code here
	            
	    }`

2. Yes, You can add the readonly key with true value.
`$this->form[] = ['label'=>'Full Name','name'=>'full_name','type'=>'text','readonly'=>true];`

**Comment by topo79** [comment]
@jdredondo @fherryfherry can I make some table content an hyperlink? like I am displaying the customers and want to click on a customer name and see a filtered view for that customer ( I am not want to have a submodule) The idea also is if I am storing URLS on the table I want to be able to click on them ( and of course to open it on a different browser window)

**Comment by jdredondo** [comment]
Yes you can do, please, this section is for reporting application issues

**Comment by topo79** [comment]
Hey @jdredondo I see on this section people asking how to questions more than reporting apps issues, if so where is the right place to ask for assistance, like my question above, and actually I was looking for a way to do it, not just the confirmation of the possibility. Don't get me wrong I have been working on CB for the past month or so and I so F.. love it! it makes everything really easy

**Comment by jdredondo** [comment]
You can use the callback function. i.e:
`$this->col[] = ["label"=>"Client","name"=>"client","callback"=>function($row) {
	if($row->client) {
		return "<a href='your_url_here'>".$row->client."</a>";
	}
}];`

**Comment by topo79** [comment]
There you go @jdredondo thank you, I will test it in a moment, now, if I want to add a row to the end of the table with totals is there a method for that or I need to edit CRUDBooster.php gettable method?

**Comment by jdredondo** [comment]
To add an additional row you would have to modify the getIndex method in the CBController.php file (not recommended), since there is no way to do this

**Comment by topo79** [comment]
Hi, @jdredondo I have a question I am trying to implement the URL, but if I do that I lose the join on the column.

This is my code:
$this->col[] = ["label"=>"Host","name"=>"host_clusterip_id","callback"=>function($row) { if($row->host_clusterip_id) { return "<a href='your_url_here'>".$row->host_clusterip_id."</a>"; } }];

and this is the result:

![image](https://user-images.githubusercontent.com/30064865/30505965-15d84f9e-9a46-11e7-82a2-553c1c805e5a.png)

and it should be a name not the id of the table.. can you point me to the right direction, please.

**Comment by harimayco** [comment]
Hi @topo79 CB uses format {join_table_name}_{column_name} when joining a table. 
so please try $row->hosts_clusterip_host_name

**Comment by topo79** [comment]
Hey @harimayco it worked like a charm!! TY!

**Comment by topo79** [comment]
@jdredondo RE:

> To add an additional row you would have to modify the getIndex method in the CBController.php file (not recommended), since there is no way to do this
> 
I need to add a total of the listed on the table ie. I have a table that has unit prices per line and I want to get a total at the end of that column, so the only way to do it is by editing the getindex? if so any idea what lines I should take a look?

Also what if I want to add the totals to an statistic, how can I make this update when I filer

**Comment by topo79** [comment]
@jdredondo Hello, its me again, in relationship with this topic how can I add an external URL like my table has company name and company URL so if I add www.google.com I want to be able to click on that in the table and get that opened on a separated window. Please help TY
